pages/lua_manual/api_reference.md

```diff
@@ -0,0 +1,1961 @@
+
+
+### Metafunctions
+
+| Metafunctions |
+| ------------- |
+| __add |
+| __eq |
+| __mul |
+
+
+### Property Descriptions
+
+####  `integer`{: .lua-type } <a name='Colourb-alpha'>alpha</a>{: .lua-method }
+
+
+
+####  `integer`{: .lua-type } <a name='Colourb-blue'>blue</a>{: .lua-method }
+
+
+
+####  `integer`{: .lua-type } <a name='Colourb-green'>green</a>{: .lua-method }
+
+
+
+####  `integer`{: .lua-type } <a name='Colourb-red'>red</a>{: .lua-method }
+
+
+
+####  `integer`{: .lua-type }, `integer`{: .lua-type }, `integer`{: .lua-type }, `integer`{: .lua-type } <a name='Colourb-rgba'>rgba</a>{: .lua-method }
+
+
+
+
+
+### Method Descriptions
+
+#### `nil`{: .lua-type} <a name='Colourb-new'>new</a>{: .lua-method }()
+
+
+
+
+
+---
+
+## <a href='#Colourf' name='Colourf'>Colourf</a>
+
+Inherits: `nil`{: .lua-type }
+
+
+
+### Properties
+
+| Name | Type |
+| ------------ | ---- |
+| [alpha](#Colourf-alpha){: .lua-method } | `number`{: .lua-type } |
+| [blue](#Colourf-blue){: .lua-method } | `number`{: .lua-type } |
+| [green](#Colourf-green){: .lua-method } | `number`{: .lua-type } |
+| [red](#Colourf-red){: .lua-method } | `number`{: .lua-type } |
+| [rgba](#Colourf-rgba){: .lua-method } | `number, number, number, number`{: .lua-type } |
+
+
+### Methods
+
+| Return Type | Name |
+| ------------ | ---- |
+| `Colourf`{: .lua-type} | [new](#Colourf-new){: .lua-method }() |
+
+
+### Metafunctions
+
+| Metafunctions |
+| ------------- |
+| __eq |
+
+
+### Property Descriptions
+
+####  `number`{: .lua-type } <a name='Colourf-alpha'>alpha</a>{: .lua-method }
+
+
+
+####  `number`{: .lua-type } <a name='Colourf-blue'>blue</a>{: .lua-method }
+
+
+
+####  `number`{: .lua-type } <a name='Colourf-green'>green</a>{: .lua-method }
+
+
+
+####  `number`{: .lua-type } <a name='Colourf-red'>red</a>{: .lua-method }
+
+
+
+####  `number`{: .lua-type }, `number`{: .lua-type }, `number`{: .lua-type }, `number`{: .lua-type } <a name='Colourf-rgba'>rgba</a>{: .lua-method }
+
+
+
+
+
+### Method Descriptions
+
+#### `nil`{: .lua-type} <a name='Colourf-new'>new</a>{: .lua-method }()
+
+
+
+
+
+---
+
+## <a href='#Context' name='Context'>Context</a>
+
+Inherits: `nil`{: .lua-type }
+
+The Context class has no constructor; it must be instantiated through the CreateContext() function. It has the following methods and properties:
+
+### Properties
+
+| Name | Type |
+| ------------ | ---- |
+| [dimensions](#Context-dimensions){: .lua-method } | `Vector2i`{: .lua-type } |
+| [documents](#Context-documents){: .lua-method } | `ContextDocumentsProxy`{: .lua-type } |
+| [focus_element](#Context-focus_element){: .lua-method } | `Element`{: .lua-type } |
+| [hover_element](#Context-hover_element){: .lua-method } | `Element`{: .lua-type } |
+| [name](#Context-name){: .lua-method } | `string`{: .lua-type } |
+| [root_element](#Context-root_element){: .lua-method } | `Element`{: .lua-type } |
+
+
+### Methods
+
+| Return Type | Name |
```

Comment:
Maybe put the return types at the end here as well, for consistency with the properties?

Reply:
Alright will do this

Reply:
Actually I also now want to change "Method" to "Function" everywhere. The term sort of carried over from the Python terminology and Lua uses the word 'function' for it

Reply:
Sounds good to me.